dockerfiles/bash/Dockerfile

```diff
@@ -3,24 +3,87 @@ WORKDIR /go/src/github.com/sourcegraph/lsp-adapter
 COPY . .
 RUN CGO_ENABLED=0 GOBIN=/usr/local/bin go install github.com/sourcegraph/lsp-adapter
 
-# 👀 Add steps here to build the language server itself 👀
-# CMD ["echo", "🚨 This statement should be removed once you have added the logic to start up the language server! 🚨 Exiting..."]
+FROM mongo:3.7.9
 
-FROM node:9
+RUN apt-get update \
+  && apt-get install --no-install-recommends -y curl=7.47.0-1ubuntu2.8 \
+  && rm -rf /var/lib/apt/lists/*
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash -
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+RUN apt-get update \
+  && apt-get install --no-install-recommends -y \
+    nodejs=8.11.2-1nodesource1 \
+    man-db=2.7.5-1 \
+    netcat=1.10-41 \
+    python=2.7.12-1~16.04 \
+    python-pip=8.1.1-2ubuntu0.4 \
+    git=1:2.7.4-0ubuntu1.3 \
+    yarn=1.7.0-1 \
+  && rm -rf /var/lib/apt/lists/*
+
+# Downgrade pip because explainshell depends on this API.
+RUN pip install --upgrade pip==9.0.3
+
+# This should be changed to the upstream https://github.com/idank/explainshell
+# once https://github.com/idank/explainshell/pull/125 is merged in.
+#
+# Ignore the warning about using WORKDIR instead of cd for convenience.
+# hadolint ignore=DL3003
+RUN git clone https://github.com/chrismwendt/explainshell \
+  && cd explainshell \
+  && git checkout 775a6097ab19ec9ee320919f7f29f7041513125e \
+  && rm -rf explainshell/.git \
+  && pip install setuptools==39.2.0 \
```

Comment:
Can this be a separate build step? (i.e. have two steps - one fetches explain shell, the other builds the project)

Reply:
I heard that it's best to reduce the number of RUN commands in a Dockerfile